scripts/xtensa-build-zephyr.py

```diff
@@ -899,6 +901,60 @@ def build_platforms():
 			symlinks=True, ignore_dangling_symlinks=True, dirs_exist_ok=True)
 
 
+def install_lib(platform, sof_output_dir, abs_build_dir):
+	global signing_key
+	with os.scandir(str(abs_build_dir)) as iter:
+		for entry in iter:
+			if (not entry.is_dir or
+			    not entry.name.endswith('_llext')):
+				continue
+			uuids = pathlib.Path(entry.path) / 'llext.uuid'
+			if not os.path.exists(uuids):
+				continue
+
+			# replace '_llext' with '.llext', e.g.
+			# eq_iir_llext/eq_iir.llext
+			llext_base = entry.name[:-6]
+			llext_file = llext_base + '.llext'
+
+			sof_lib_dir = pathlib.Path(sof_output_dir / '..' / 'sof-ipc4-lib' / platform)
+			if args.key_type_subdir != "none":
+				sof_lib_dir = sof_lib_dir / args.key_type_subdir
+			sof_lib_dir.mkdir(parents=True, exist_ok=True)
+			dst = sof_lib_dir / llext_file
+			rimage_executable = shutil.which("rimage", path=RIMAGE_BUILD_DIR)
+			rimage_cfg = pathlib.Path(entry.path) / 'rimage_config.toml'
+			llext_input = pathlib.Path(entry.path) / str(llext_base + '.so')
+			llext_output = pathlib.Path(entry.path) / llext_file
+
+			sign_cmd = [str(rimage_executable), "-o", str(llext_output),
+				    "-e", "-c", str(rimage_cfg),
+				    "-k", str(signing_key), "-l", "-r",
+				    str(llext_input)]
+			execute_command(sign_cmd, cwd=west_top)
+
+			try:
+				fdst = open(str(dst), 'wb')
+				fllext = open(str(llext_output), 'rb')
+				fman = open(str(llext_output) + '.xman', 'rb')
+			except:
+				sys.exit("Failed to open files")
+
+			shutil.copyfileobj(fman, fdst)
+			shutil.copyfileobj(fllext, fdst)
+
+			fdst.close()
+			fman.close()
+			fllext.close()
```

Comment:
```python
with open('file1.txt', 'r') as file1, open('file2.txt', 'r') as file2:
```

Reply:
> Or you can just leave them open, they will be automatically closed when the script exits.

@marc-hb it's a loop over potentially many modules, would be dirty to leave them all open, particularly with their handle variables re-used

Reply:
You could have a single `with` (and indentation level) for all these files at once.

Reply:
@marc-hb yes, you proposed it above, but if any of them fails, we don't need to continue and should abort, whereas `with` would "hide" the error?